doc/user/content/overview/architecture.md

```diff
@@ -8,23 +8,124 @@ menu:
+
+#### Sinks
+
+Sinks are the inverse of sources and represent a connection to an external stream where Materialize outputs data from its persistent storage. Sinks don't read directly from the compute layer [todo: is this true?]. Sinks, like sources begin their works immediately. When a user defines a sink over a materialized view, source, or table, Materialize automatically generates the required schema and writes down the stream of changes to that materialized view, source, or table. The amount of computational resources (CPU and memory) available to encode all of the data that needs to be written can be scaled up and down by changing the sink's size with `ALTER SINK`.
+
 ### Compute
 
-### Adapter
+Compute is responsible for reading in input collections from `STORAGE` (by reading from S3 instead of the upstream external system) [todo: is this too prescriptive? I want to get across that compute doesn't touch kafka], and transforming those collections into ouput collections that contain correct answers for the requested computations.
+
+The key concepts and nouns that users have to interact with in are `VIEW`s, `CLUSTER`s, `CLUSTER REPLICA`s, `INDEX`es, and `MATERIALIZED VIEW`s.
+
+todo: we need a visual here describing the potential relationships between these objects
+todo: I wish we could talk about the catalog bc I think that would make things more clear. "logical" objects only live in the catalog, whereas physical objects correspond to something being spun up in aws or in memory on a aws instance
+
+#### (Unmaterialized) Views
+
+In SQL, unmaterialized views represent a query that you save with some given name. These are used primarily as shorthand for some lengthy, complicated SELECT statement. Materialize uses the idiom of views similarly for unmaterialized views. Like connections, creating a view doesn't cause any computation to take place. Only selecting from a view or creating an index on a view causes actual computation to happen. Finally, note that a materialized view is very different from a unmaterialized view and we will have more to say about it shortly.
+
+Unmaterialized views can be referenced on any cluster in the region.
+
+#### Clusters
+
+Clusters are logical components that let you express resource isolation for all objects responsible for performing and maintaining views e.g. indexes and materialized views. When creating these objects, you must specify which cluster you want to use. (Not explicitly naming a cluster uses your session’s default cluster.) But, a cluster is a purely logical component that exists to indicate which computations can be bundled together on the same physical resources, and computation only begins when we create a cluster replica.
+
+[todo: operational concerns for clusters].
```

Comment:
We can leave operational concerns out of this page; these will be covered in the context of "user workflows" and more operations-oriented docs.